RepairsListener/UseCase/AddRepairsContractsUseCase.cs

```diff
@@ -0,0 +1,63 @@
+using RepairsListener.Boundary;
+using RepairsListener.Gateway.Interfaces;
+using RepairsListener.UseCase.Interfaces;
+using Hackney.Core.Logging;
+using System;
+using System.Threading.Tasks;
+using Hackney.Shared.Asset.Domain;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using RepairsListener.Infrastructure;
+
+namespace RepairsListener.UseCase
+{
+    public class AddRepairsContractsUseCase : IAddRepairsContractsUseCase
+    {
+        private readonly IRepairsStoredProcedureGateway _gateway;
+        private readonly ILogger<IAddRepairsContractsUseCase> _logger;
+
+
+        public AddRepairsContractsUseCase(IRepairsStoredProcedureGateway gateway, ILogger<IAddRepairsContractsUseCase> logger) : base()
+        {
+            _gateway = gateway;
+            _logger = logger;
+        }
+
+        [LogCall]
+        public async Task ProcessMessageAsync(EntityEventSns message)
+        {
+            if (message is null) throw new ArgumentNullException(nameof(message));
+
+            string jsonSnsMessage = JsonSerializer.Serialize(message);
+            _logger.LogInformation("RepairsListener received AssetCreatedEvent SNS message with body {JsonSnsMessage}", jsonSnsMessage);
+
+            // Get asset data from SNS message
+            string addRepairsContractsMessage = message.EventData.NewData.ToString();
```

Comment:
Add a null check for new data - don't throw an exception, just log it out and quit

Reply:
I can log and "return". Out of curiosity why not throw an exception?

Reply:
Like this? 
![image](https://github.com/LBHackney-IT/repairs-listener/assets/70756861/c803c727-0d76-451b-9845-49720facc7d4)

Reply:
To be fair, you make a good point - throwing an exception would ensure that malformed messages are investigated. I was going to go with the angle "because it's not an issue within the listener" but it actually would be a situation worth investigating - so you have changed my mind